wal.go

```diff
@@ -411,6 +415,20 @@ func (w *WAL) GetSealedLogFiles(fromIndex uint64) ([]*SealedSegmentInfo, error)
 	return sealedSegInfo, nil
 }
 
+func (w *WAL) MarkSealedLogDeletable(minIndex uint64) error {
```

Comment:
This seems that this only works if the minIndex (which is baseIndex) is given. Could there be a case where we don't know necessarily what is the minIndex/baseIndex and yet we want to delete a segment which contains Index `X` even if `X` is not the min/base Index? 
For that, we probably need to use `findSegmentReader` api. 
If we go with that we could avoid a another API call to get the base index and then try to mark it as deletable.

Reply:
I was assuming that this interface is to be used externally, feeding in information obtained by GetSealedLogFiles, which contain the baseIndex.

Reply:
Ah, make sense!